engine/src/cmd/movable.cpp

```diff
@@ -0,0 +1,432 @@
+    this->last_processed_sqs = cur_sim_frame;
+    this->cur_sim_queue_slot = (cur_sim_frame+this->sim_atom_multiplier)%SIM_QUEUE_SIZE;
+    Transformation old_physical_state = curr_physical_state;
+
+    UpdatePhysics3(trans, transmat, lastframe, uc, superunit);
+
+
+
+
+    if (resolveforces) {
+        //clamp velocity
+        ResolveForces(trans, transmat);
+        if (Velocity.i > VELOCITY_MAX)
+            Velocity.i = VELOCITY_MAX;
+
+        else if (Velocity.i < -VELOCITY_MAX)
+            Velocity.i = -VELOCITY_MAX;
+        if (Velocity.j > VELOCITY_MAX)
+            Velocity.j = VELOCITY_MAX;
+
+        else if (Velocity.j < -VELOCITY_MAX)
+            Velocity.j = -VELOCITY_MAX;
+        if (Velocity.k > VELOCITY_MAX)
+            Velocity.k = VELOCITY_MAX;
+
+        else if (Velocity.k < -VELOCITY_MAX)
+            Velocity.k = -VELOCITY_MAX;
+    }
+
+    // The 1.0 difficulty is a hack based on the hack in GetVelocityDifficultyMult
+    this->UpdatePhysics2( trans, old_physical_state, Vector(), 1.0, transmat, cum_vel, lastframe, uc );
+
+
+}
+
+void Movable::AddVelocity( float difficulty )
+{
+    //for the heck of it.
+    static float humanwarprampuptime = XMLSupport::parse_float( vs_config->getVariable( "physics", "warprampuptime", "5" ) );
+    //for the heck of it.
+    static float compwarprampuptime  =
+        XMLSupport::parse_float( vs_config->getVariable( "physics", "computerwarprampuptime", "10" ) );
+    static float warprampdowntime    = XMLSupport::parse_float( vs_config->getVariable( "physics", "warprampdowntime", "0.5" ) );
+    float  lastWarpField = graphicOptions.WarpFieldStrength;
+
+    bool   playa = isPlayerShip();
+
+    float  warprampuptime = playa ? humanwarprampuptime : compwarprampuptime;
+    //Warp Turning on/off
+    if (graphicOptions.WarpRamping) {
+        float  oldrampcounter = graphicOptions.RampCounter;
+        if (graphicOptions.InWarp == 1)             //Warp Turning on
+            graphicOptions.RampCounter = warprampuptime;
+        else                                        //Warp Turning off
+            graphicOptions.RampCounter = warprampdowntime;
+        //switched mid - ramp time; we also know old mode's ramptime != 0, or there won't be ramping
+        if (oldrampcounter != 0 && graphicOptions.RampCounter != 0 ) {
+            if (graphicOptions.InWarp == 1)             //Warp is turning on before it turned off
+                graphicOptions.RampCounter *= (1 - oldrampcounter/warprampdowntime);
+            else                                        //Warp is turning off before it turned on
+                graphicOptions.RampCounter *= (1 - oldrampcounter/warprampuptime);
+        }
+        graphicOptions.WarpRamping = 0;
+    }
+    if (graphicOptions.InWarp == 1 || graphicOptions.RampCounter != 0) {
+        float rampmult = 1.f;
+        if (graphicOptions.RampCounter != 0) {
+            graphicOptions.RampCounter -= SIMULATION_ATOM;
```

Comment:
I think this is actually supposed to be `simulation_atom_var`

Reply:
Fixed.